src/List.h

```diff
@@ -29,7 +29,9 @@
 // TODO: this can be removed in v3.1 when List::sort() is removed
 typedef int (*list_cmp_func)(const void* v1, const void* v2);
 
-template<typename T>
+enum list_order { LIST_ORDERED, LIST_UNORDERED };
```

Comment:
```suggestion
enum ListOrder { LIST_ORDERED, LIST_UNORDERED };
```

Think that's the naming convention.

`enum class` may also be preferable, in general.

Reply:
I was following the naming scheme used by `dict_order` there, but that could probably be renamed too.